### DISCUSSION TITLE
Welcome to AutoDashBackEnd Discussions!

### DISCUSSION THREAD

**Discussion by Mathews2115** [Mathews2115/AutoDashBackEnd#1] [Announcements]
## 👋 Yo!

**Comment by Mathews2115** [comment]
Hey! Sorry I missed this.

I have the Terminator EFI
<https://www.holley.com/products/fuel_systems/fuel_injection/terminator_efi/terminator_4bbl_tbi_kits/4bbl_tbi_kits/parts/550-406>ECU
as well and I used their tuning software found here
<https://www.holley.com/support/fuel_injection/hp_and_dominator_efi/>.

For me, I was able to find the option to change the CAN #1 and #2 outputs
in the Holley EFI software - I believe it is in the BASIC I/O tab (from the
System Parameters that lines up on the left) - and then the tab it opens
up, there will be an option to configure the CANBUS outputs (Holley or
NHRA's supported Racepak).  It's a little buried but is there, just make
sure you have the most updated version.

And originally I was confused because the software says it is for
HP/Dominator ecus but then I read this line about my Terminator:
>Terminator ECU can be EASILY updated to Holley HP ECU specs with a free
online software download. ECU can then be used for LSX, boosted, nitrous,
and many other applications in the future as you desire!

I guess that explains why it just worked (just in case you hit the same
snag I did!)

Let me know if you have any other questions!

**Reply by 350SBC** [reply]
The v3 updates allow you to use the j3 connector for can out in racepak format. 
Just ran a capture from it and it is functional.
I bought the connector pins and plugs from digikey for $34 with shipping. 
or you an go to holley's overpriced connectors. 
I have pn's if needed. 

Deciphering the standard holley code is not for the faint of heart.

**Comment by 350SBC** [comment]
I am having trouble figuring out a few things on this. 
First where are you setting the resolution/canvas size? i would like to use a different screen.
I also cant get the system you built to allow the use of a hostname vs localhost.
it looks like you are handing the URL to the DashSocketComms.js however it does not use it. 
I was looking at using a few Pis to run displays and one for the CAN interface.

**Reply by 350SBC** [reply]
The resolution related pieces are in the fronted. Easy enough.
I dont really understand how these pieces fit together.
Did you bundle a "Compiled" version of the frontend o the backend repo?
Also did you use a int16 for the odometer?  that is  ~ -32K to +32K so 64000 miles. but the display format support 99999? 
If i get am understanding i may make some changes to better match the type to their uses.  
I plan to use this on my gmc k2500 4x4 its on the third lap of the 99K odometer. so more odometer would be nice.

**Comment by 350SBC** [comment]
Made a couple changes to support a higher range for the odometer. for both the backend ad frontend that is bundled with it 
Would like to push them to a branch and put in a pull request. Can you give me right to create/push? Would prefer that over forking.

**Reply by Mathews2115** [reply]
Hey! Sorry it took me so damn long to get back to you, I didnt realize I locked down everything and prevented PRs. I'll open up to PRs asap today.

Sounds like you got a lot figured out - yeah, I compile the FrontEnd, and drop it into that backend directory. 
I also have a "boilerplate frontend" as well if you havent seen that - worth checking out.

**Reply by 350SBC** [reply]
No issues. Thanks

**Reply by 350SBC** [reply]
What was the strategy behind treating GPS speed differently than CAN speed?
Does the odometer function rely on GPS odometry? 
Have you had any issues with the updateMPG section. I am getting an out of range exception  and the backend crashes when i get  out of throttle at around 75 mph.  stepping through the section while the candump plays prevents the exception.

**Comment by Mathews2115** [comment]
Oh, no strategy at all! My build doesn't have any speed via CAN (no speed
going through Holley) so I used GPS to handle it instead. also, my fuel
sender was broken so I ended up doing a calculation from the fuel flow to
replace it - if your sender works, id highly recommend plumbing and that
into the UI instead of trusting my math 😁

The odometer function is also purely based on the gps.

As far as the big you're running into, negative, have not seen that but I
don't mind helping debugging it!   What's you're hardware setup?

**Reply by 350SBC** [reply]
4 Gb pi4, Usb2CAn from amazon and a 10.1 inch display . 
This is being used on my 79 Gmc 4x4 Truck which is running a 350 SBC with a 8-71 Weiend and some edlebrock 215 heads, Holley Terminator x max, 4l80 transmission, and a np 205 Transfer case.
The terminator X max is running Dual stealth Throttle bodies, and the transmission.  
I am unable to create a branch on your repo, I created a fork https://github.com/350SBC/AutoDashBackEnd.
There are three new can_dumps on the https://github.com/350SBC/AutoDashBackEnd/tree/CAN_ONLY_SPEEDO_FRONT_END branch. 
The 20th had the original can_dump, i wanted to more quickly see the error so is removed a most of the time around the error "candump-edited_for_error.log". 
Line 141 of ecumanger.js is the line that I commented to bypass the error. Uncomment it and play the candump for roughly 60 seconds to reproduce the error.
There are a few other mods to the packages file a few lines to app.js to be run without the front end if the NODE_ENV is not production.
